src/axom/primal/operators/detail/intersect_bounding_box_impl.hpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef PRIMAL_INTERSECT_BOUNDING_BOX_HPP_
+#define PRIMAL_INTERSECT_BOUNDING_BOX_HPP_
+
+#include "axom/core/Macros.hpp"
+
+#include <type_traits> // for std::is_floating_point< T >()
+
+namespace axom
+{
+namespace primal
+{
+namespace detail
+{
+
+/*!
+ * \brief Helper routine for AABB / AABB intersection test.
+ *
+ * \param [in] min1 the 1st AABB min coordinate along a direction.
+ * \param [in] max1 the 1st AABB max coordinate along a drection.
+ * \param [in] min2 the 2nd AABB min coordinate along a direction.
+ * \param [in] max2 the 2nd AABB max coordinate along a drection.
+ *
+ * \return status true if the AABB's intersect, otherwise false.
+ *
+ * \note This routine is called by the intersect AABB/AABB methods for each
+ *  spatial dimension.
+ */
+template < typename T >
+AXOM_HOST_DEVICE
+inline bool intersect_bbox_bbox_test( const T& min1,
```

Comment:
`BoundingBox::intersectsWith()` should also call this method now -- This would avoid duplicating code and also expose this method to the `BoundingBox` tests

Reply:
Done!